great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -374,36 +383,474 @@ def apply_theme(
+                        batch_component.name,
+                        type=batch_component.alt_type,
+                        title=None,
+                        axis=alt.Axis(labels=False, ticks=False, grid=False),
+                    ),
+                    y=alt.Y(
+                        column_number_component.name,
+                        type=column_number_component.alt_type,
+                        title=" ",
+                        axis=alt.Axis(labels=False, ticks=False),
+                    ),
+                )
+            ).mark_text(
+                text=text,
+                color=Colors.PURPLE.value,
+                lineBreak=r"$",
+                dy=dy,
+            )
+
+        return chart
+
+    @staticmethod
+    def _get_nonsequential_isotype_chart(
+        df: pd.DataFrame,
+        metric_component: MetricPlotComponent,
+        batch_component: BatchPlotComponent,
+        domain_component: DomainPlotComponent,
+        column_number_component: PlotComponent,
+        column_set: Optional[List[str]],
+    ) -> alt.Chart:
+        title: alt.TitleParams = determine_plot_title(
+            metric_plot_component=metric_component,
+            batch_plot_component=batch_component,
+            domain_plot_component=domain_component,
+        )
+
+        tooltip: List[alt.Tooltip] = batch_component.generate_tooltip() + [
+            metric_component.generate_tooltip(),
+        ]
+
+        chart: alt.Chart
+        if column_set is None:
+            chart = (
+                alt.Chart(data=df, title=title)
+                .mark_point(color=Colors.PURPLE.value)
+                .encode(
+                    x=alt.X(
+                        batch_component.name,
+                        type=batch_component.alt_type,
+                        title=batch_component.title,
+                        axis=alt.Axis(labels=False, grid=False),
+                    ),
+                    y=alt.Y(
+                        column_number_component.name,
+                        type=column_number_component.alt_type,
+                        title=column_number_component.title,
+                        axis=alt.Axis(grid=True),
+                    ),
+                    tooltip=tooltip,
+                )
+            )
+        else:
+            dy: int
+            if len(column_set) > 50:
+                text = f"All batches have the same set of columns. The number of columns ({len(column_set)}) is too long to list here."
+                dy = 0
+            else:
+                column_set_text: str = ""
+                idx: int = 1
+                for column in column_set:
+                    if idx % 4 == 0:
+                        column_set_text += f"{column},$"
+                    else:
+                        column_set_text += f"{column}, "
+                    idx += 1
+                text = f"All batches have columns matching the set:${column_set_text[:-2]}."
+                dy = -100
```

Comment:
Looks like much of this logic is duplicated from before - perhaps a chance for a refactor? (not necessary for this PR)